src/plugins/gui/scenes/AddressFormScene.tsx

```diff
@@ -235,6 +235,14 @@ export const AddressFormScene = React.memo((props: Props) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [searchResults])
 
+  // Unmount cleanup
+  React.useEffect(
+    () => () => {
+      onClose()
+    },
+    [onClose]
```

Comment:
If the caller passes a different `onClose` instance, this hook will call the previous `onClose` instance before installing the new instance, even if the scene is still mounted. This is kinda unsafe (even though we don't have a way to trigger the bug currently).

I would have expected this to be attached to the navigation event, like this:

```ts
React.useEffect(() => {
    return navigation.addListener('beforeRemove', () => {
      if (onClose != null) onClose()
    });
  },
  [navigation, onClose]
);
```

This will correctly handle `onClose` being updated, but without firing any callbacks until the scene is *actually* removed.

Same for SepaFormScene